tdrs-backend/tdpservice/search_indexes/parsers/preparser.py

```diff
@@ -0,0 +1,262 @@
+"""Converts data files into a model that can be indexed by Elasticsearch."""
+
+import os
+import logging
+import argparse
+from cerberus import Validator
+from .util import get_record_type
+from . import tanf_parser
+
+from io import BufferedReader
+# from .models import ParserLog
+from tdpservice.data_files.models import DataFile
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+
+def validate_header(row, data_type, given_section):
+    """Validate the header line of the datafile."""
+    """
+    https://www.acf.hhs.gov/sites/default/files/documents/ofa/transmission_file_header_trailer_record.pdf
+
+    DESCRIPTION		LENGTH	FROM	TO	COMMENT
+    Title		    6	1	6	Value	=	HEADER
+    YYYYQ	        5	7	11	Value	=	YYYYQ
+
+    Type 	        1	12	12	A=Active;	C=Closed;	G=Aggregate,	S=Stratum
+    State Fips	    2	13	14	"2	digit	state	code	000	a	tribe"
+    Tribe Code	    3	15	17	"3	digit	tribe	code	000	a	state"
+    Program	Type	3	18	20	Value	=	TAN	(TANF)	or	Value	=	SSP	(SSP-MOE)
+    Edit Indicator	1	21	21	1=Return	Fatal	&	Warning	Edits	2=Return	Fatal	Edits	only
+    Encryption      1	22	22	E=SSN	is	encrypted	Blank	=	SSN	is	not	encrypted
+    Update      	1	23	23	N	=	New	data	D	=	Delete	existing	data	U
+    QUARTERS:
+        Q=1	(Jan-Mar)
+        Q=2	(Apr-Jun)
+        Q=3	(Jul-Sep)
+        Q=4	(Oct-Dec)
+    Example:
+    HEADERYYYYQTFIPSSP1EN
+    """
+
+    logger.debug('Validating header row.')
+    section_map = {
+        'A': 'Active Case Data',
+        'C': 'Closed Case Data',
+        'G': 'Aggregate Data',
+        'S': 'Stratum Data',
+    }
+
+    try:
+        header = {
+            'title':        row[0:6],
+            'year':         row[6:10],
+            'quarter':      row[10:11],
+            'type':         row[11:12],
+            'state_fips':   row[12:14],
+            'tribe_code':   row[14:17],
+            'program_type': row[17:20],
+            'edit':         row[20:21],
+            'encryption':   row[21:22],
+            'update':       row[22:23],
+        }
+
+        for key, value in header.items():
+            logger.debug('Header key %s: "%s"' % (key, value))
+
+        # TODO: Will need to be saved in parserLog, #1354
+
+        try:
+            # logger.debug("Given section: '%s'\t Header section: '%s'", given_section, section_map[header['type']])
+            # logger.debug("Given program type: '%s'\t Header program type: '%s'", data_type, header['program_type'])
+
+            if given_section != section_map[header['type']]:
+                raise ValueError('Given section does not match header section.')
+            if (data_type == 'TANF' and header['program_type'] != 'TAN')\
+                    or (data_type == 'SSP' and header['program_type'] != 'SSP')\
+                    or (data_type not in ['TANF', 'SSP']):
+                raise ValueError("Given data type does not match header program type.")
```

Comment:
It looks like there are three ways this can can be raised. Can the reason of the value error be put in the error log?

Reply:
I have re-commented in the debug lines on lines 71, 72. I think it was causing an issue on CircleCI with the large files logging every line. For non-local testing, I think we might need to turn our logging level down to "INFO" however. I think this will be a team-wide discussion including @ADPennington

Reply:
Team decided that prod/staging should use info.